internal/cluster/cluster_kvstore.go

```diff
@@ -43,6 +49,11 @@ func (c *Cluster) Store(client consul.Client) error {
 func Load(client consul.Client) (map[string]*Cluster, error) {
 	var clusters = map[string]*Cluster{}
 
+	err := client.LockWaitReleasead(consul.KvClustersPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not lock the kv for clusters")
```

Comment:
this message is a bit misleading: we not trying to acquire a lock here, we're just waiting for a lock to be released.